src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java

```diff
@@ -65,28 +82,30 @@ public void processContributionsForStages(Set<Stage> stages) {
         }
     }
 
-    Optional<LocalDate> calculateDueContributionDate(Set<SomuItem> contributionSomuItems) {
+    Optional<LocalDate> calculateDueContributionDate(Set<Contribution> contributionSomuItems) {
         return contributionSomuItems
                 .stream()
-                .filter(csi -> !(new ContributionSomuInspector(csi).hasContributionStatus()))
-                .map(csi -> new ContributionSomuInspector(csi).getContributionDueLocalDate())
+                .filter(contribution -> contribution.getStatus() == NONE)
+                .map(Contribution::getDueDate)
                 .sorted()
                 .findFirst();
     }
 
-    Optional<ContributionStatus> highestContributionStatus(Set<SomuItem> contributionSomuItems) {
-        return highestContributionStatus(contributionSomuItems, LocalDate.now());
+    Optional<Contribution.ContributionStatus> highestContributionStatus(Set<Contribution> contributions) {
+        return highestContributionStatus(contributions, LocalDate.now());
     }
 
-    Optional<ContributionStatus> highestContributionStatus(Set<SomuItem> contributionSomuItems, LocalDate now) {
+    Optional<Contribution.ContributionStatus> highestContributionStatus(Set<Contribution> contributionSomuItems, LocalDate now) {
         return contributionSomuItems
                 .stream()
                 .map(csi -> {
-                    ContributionStatus contributionStatus = new ContributionSomuInspector(csi).getContributionStatus();
-                    if (contributionStatus.equals(CONTRIBUTION_RECEIVED) || contributionStatus.equals(CONTRIBUTION_CANCELLED)) {
-                        return contributionStatus;
+                    Contribution.ContributionStatus contributionStatus = csi.getStatus();
+                    if (contributionStatus.equals(CONTRIBUTION_RECEIVED)) {
+                        return CONTRIBUTION_RECEIVED;
+                    } else if (contributionStatus.equals(CONTRIBUTION_CANCELLED)) {
```

Comment:
not sure on the benefit of splitting these into their own if elses? Returning the contributionStatus does the same thing I think

Reply:
Yeah good point, got overlooked...